xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/Wrappers.java

```diff
@@ -247,12 +322,16 @@ public XrpCurrencyAmount minus(XrpCurrencyAmount other) {
      * @return The product of this amount and the {@code other} amount, as an {@link XrpCurrencyAmount}.
      */
     public XrpCurrencyAmount times(XrpCurrencyAmount other) {
-      return XrpCurrencyAmount.of(this.value().times(other.value()));
+      return XrpCurrencyAmount.ofDrops(this.value().times(other.value()), this.isNegative() | other.isNegative());
     }
 
     @Override
     public String toString() {
-      return this.value().toString();
+      if (isNegative()) {
+        return "-" + this.value().toString();
```

Comment:
@nkramer44 WDYT about this one? In this case, I do this because `XrpCurrencyAmount` always holds an `UnsignedLong` (the `isNegative()` function is use to detect negativity since the Unsigned value can't hold a negative.

One could argue that this `toString` should always show a positive value, to conform with the `UnsignedLong` underlying.

Note: The details informing this choice are subtly different in the case of `IssueCurrencyAmount`.

Reply:
I think this is the right thing to do. If you deserialize one of those payloads that has a negative XRP amount and then call `toString` on that amount, it would be misleading/incorrect to _not_ show a `-` before the number IMO

Reply:
I _think_ you're right. One final counter-point here would be to emit something like `1 XRP [positive]` or `1 XRP [negative]` so that nobody accidentally uses this `toString` to show or parse real numeric values.

Thoughts?

Reply:
Resolving this for now in the interest of maintaining something as-close-as-possible to existing functionality.